### PR TITLE
[HDBDD] Set foreign keys by default in entities when none are specified

### LINKED ISSUE #618: [HDBDD] Parser Selecting default foreign keys
When an association with unmanaged foreign keys is created for example:

`
entity Address {
  key id : Integer;
  key zipCode: String(30);
};

entity Person
{
  key id : Integer;
  address1 : Association to Address;
  address2 : Association to Address { id };
};
`
For _address1_ where no elements are specified all the elements in the target(_Address_) marked with 'key' should be selected as foreign keys.

### PR DESCRIPTION
**Pull request by d-velev**
Closes #618

### CODE REVIEW

File: modules/parsers/parser-hdbdd/src/main/java/com/sap/xsk/parser/hdbdd/custom/ReferenceResolvingListener.java [hunk 1]
```diff
@@ -111,6 +116,13 @@ public void enterAssociationTarget(CdsParser.AssociationTargetContext ctx) {
           resolvedSymbol.getIdToken().getLine(), resolvedSymbol.getIdToken().getCharPositionInLine()));
     }
 
+    if (ctx.getParent().getChild(ManagedForeignKeysContext.class, 0) == null
+        && ctx.getParent().getChild(UnmanagedForeignKeyContext.class, 0) == null) {
+      List<EntityElementSymbol> associationKeys = ((EntitySymbol) resolvedSymbol).getKeys();
+      associationSymbol.setForeignKeys(associationKeys);
+      associationSymbol.setManaged(true);
```
Comment (d-velev) [inline modules/parsers/parser-hdbdd/src/main/java/com/sap/xsk/parser/hdbdd/custom/ReferenceResolvingListener.java:123]: @GrozevJunior Is this correct? We saw that the columns in the foreign key objects weren't set properly (it was just the name of the entity repeated for each column). This fixed it but maybe it's not being set at the right place.

**Review by ThuF — APPROVED** [review]
@GrozevJunior to give the final approval.

File: modules/parsers/parser-hdbdd/src/main/java/com/sap/xsk/parser/hdbdd/custom/ReferenceResolvingListener.java (hunk 1 above)
Comment (GrozevJunior) [reply modules/parsers/parser-hdbdd/src/main/java/com/sap/xsk/parser/hdbdd/custom/ReferenceResolvingListener.java:123]: @d-velev You have a point there! Then you can do it in 'public void exitAssociation(@NotNull CdsParser.AssociationContext ctx)' in this case the target will be resolved and you could take the association by the context and jsut add the foreign keys.

**Review by ThuF — APPROVED** [review]
+1

**Pull request merged**